src/DynamoApplications/StartupUtils.cs

```diff
@@ -205,6 +217,9 @@ private static void ShowHelp(OptionSet opSet)
         /// </summary>
         /// <param name="geometryFactoryPath">libG ProtoInterface path</param>
         /// <param name="preloaderLocation">libG folder path</param>
+#if NET6_0_OR_GREATER
+        [System.Runtime.Versioning.SupportedOSPlatform("windows")]
```

Comment:
Not sure if they are completely omitted by the compiler but they do generate build errors if you try to use them with an incompatible runtime identifier (i,e. <RuntimeIdentifier>linux-x64</RuntimeIdentifier> will generate compile error if you try to use a windows only API)

Reply:
`PreloadShapeManager` is windows only for now because it currently relies a lot on the windows registry to figure out where ASM is installed.
The entire logic of searching a machine for ASM installations does not really make sense for linux, since that will be mainly for services. In the case of services we are most likely going to just supply the path to ASM